utils/navigator_robotx_comms/nodes/navigator_robotx_comms.py

```diff
@@ -0,0 +1,413 @@
+                                                            self.handle_detect_deliver_message)
+
+        # start sending heartbeat every second
+        rospy.Timer(rospy.Duration(1), self.handle_heartbeat_message)
+
+    def handle_heartbeat_message(self, data):
+        hst_date_time = self.get_hst_date_time()
+        message = self.robot_x_heartbeat_message.to_string(self.delim, self.team_id, hst_date_time)
+        self.robot_x_client.send_message(message.message)
+        return message
+
+    def handle_entrance_exit_gate_message(self, data):
+        hst_date_time = self.get_hst_date_time()
+        message = self.robot_x_entrance_exit_gate_message.to_string(self.delim, self.team_id, hst_date_time, data)
+        self.robot_x_client.send_message(message.message)
+        return message
+
+    def handle_scan_code_message(self, data):
+        hst_date_time = self.get_hst_date_time()
+        message = self.robot_x_scan_code_message.to_string(self.delim, self.team_id, hst_date_time, data)
+        self.robot_x_client.send_message(message.message)
+        return message
+
+    def handle_identify_symbols_dock_message(self, data):
+        hst_date_time = self.get_hst_date_time()
+        message = self.robot_x_identify_symbols_dock_message.to_string(self.delim, self.team_id, hst_date_time, data)
+        self.robot_x_client.send_message(message.message)
+        return message
+
+    def handle_detect_deliver_message(self, data):
+        hst_date_time = self.get_hst_date_time()
+        message = self.robot_x_detect_deliver_message.to_string(self.delim, self.team_id, hst_date_time, data)
+        self.robot_x_client.send_message(message.message)
+        return message
+
+    def get_hst_date_time(self):
+        # HST is 10 hours behind UTC
+        hst_time = datetime.datetime.utcnow() - datetime.timedelta(hours=10, minutes=0)
+        date_string = hst_time.strftime("%d%m%y")
+        time_string = hst_time.strftime("%H%M%S")
+        return date_string + self.delim + time_string
+
+
+class RobotXClient():
+
+    def __init__(self):
+        self.TCP_IP = rospy.get_param('td_ip')
+        self.TCP_PORT = rospy.get_param('td_port')
+        self.BUFFER_SIZE = 1024
+        self.connected = False
+        self.socket_connection = None
+
+    def recreate_socket(self):
+        self.socket_connection = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+    def connect(self):
+        if not self.connected:
+            print('Attempting Connection to TD Server')
+        while not self.connected and not rospy.is_shutdown():
+            # recreate socket
+            self.socket_connection = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            # attempt to reconnect, otherwise sleep for 2 seconds
+            try:
+                self.recreate_socket()
+                self.connected = True
+                print('Connection to TD Server Successful')
```

Comment:
use `rospy.loginfo` so the prints are seen in the launch terminal and written to file